### PR TITLE
Prepare Ruby's Debugging lesson to move before Enumerable lessons

### LINKED ISSUE #23224: Prepare Ruby's Debugging lesson to move before Enumerable lessons
**Lesson:**
https://www.theodinproject.com/paths/full-stack-ruby-on-rails/courses/ruby-programming/lessons/debugging

**Description:**
People have a difficult time solving the TDD exercises for basic enumerables, predicate enumerables, and nested collections because they have not been introduced to debugging. We need to remove references to all enumerables from the debugging lesson, so that we can move it before these 3 lessons.

**Acceptance Criteria:**
* Replace enumerables examples
* Update the surrounding context/explanations of the debugging examples
* Add a link to documentation when explaining why a method is not working as expected.

**Additional Information:**
Below is an idea that I had to replace the `.map` examples, however you are welcome to come up with your own example that does not produce an error (and does not return the correct answer).

For this example, an isogram is a word that has no repeating letters (consecutive or nonconsecutive).
```rb
def isogram?(string)
  original_length = string.length
  string_array = string.downcase.split
  unique_length = string_array.uniq.length
  original_length == unique_length
end

isogram?("Odin")
```
Source: https://www.codewars.com/kata/54ba84be607a92aa900000f1

**NOTES:**
* Comment on this issue to indicate that you would like to work on it.
* Do **not** start to work on this until you have been assigned to it. 
* You must read and follow our [contribution guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by Maheshkumar-novice**
- [x] You have read [The Odin Projects Contributing Guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md).
 - [x] The title summarizes the change and where it happened, for example: "Fixes punctuation in Clean Code lesson".
 - [x] If the PR is related to an open issue, use a [relevant keyword](https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests) and reference it with the `#` sign and the issue number.
 - [x] You have previewed your markdown (if any) in the [Odin Markdown Preview Tool](https://www.theodinproject.com/lessons/preview)
 - [x] If changes are requested, please make the changes in a timely manner. After you submit the changes, request another review from the maintainer (top right).

#### 1. Describe the changes made and include why they are necessary or important:

Prepare Ruby's Debugging lesson to move before Enumerable lessons

- [x] Replaced enumerables examples
- [x] Updated the surrounding context/explanations of the debugging examples
- [x] Added a link to documentation when explaining why a method is not working as expected.

#### 2. Related Issue

Closes #23224

### CODE REVIEW

**Comment by rlmoser99** [comment]
Wow, thanks for updating this lesson so quickly! I will take a closer look at this in the few days. 👍

**Comment by Maheshkumar-novice** [comment]
Happy to help. Sure. Thanks for the **isogram** code!

File: ruby_programming/basic_ruby/debugging.md [hunk 1]
```diff
@@ -113,16 +116,17 @@ To follow along with these examples save the code into a Ruby file (e.g., `scrip
...
-  string_array.join(' ')
+  unique_length = string_array.uniq.length
+  original_length == unique_length
 end
 
-double_words_in_phrase("This is a test")
+isogram?("Odin")
 ~~~
 
 When your code executes and gets to `binding.pry`, it will open an IRB-like session in your terminal. You can then use that session to check the values of anything within the scope of where you included `binding.pry`. However, keep in mind that any code written *after* the `binding.pry` statement will not have been evaluated during the Pry session. Thus, adding a `binding.pry` line in our code is similar to creating a breakpoint in JavaScript.
```
Comment (rlmoser99) [inline ruby_programming/basic_ruby/debugging.md:132]: "anything within the scope"
I know this was part of the original lesson, but I think this wording is misleading. What do you think of clarifying this and maybe adding specific examples of what values will be available (`original_length` and `string_array`, but not `unique_length`)? I think this paragraph would be a bit more meaningful with examples.

**Review by rlmoser99 — CHANGES_REQUESTED** [review]
Thank you so much for taking the time to work on this lesson. Sorry that it took me a while to look at it, not only did we work on a home remodel during the holidays, but I also got sick. Let me know what you think of my one suggestion, as it was not part of the original issue. Just something that stuck out to me as I was re-reading the entire lesson.

**Comment by Maheshkumar-novice** [comment]
> Thank you so much for taking the time to work on this lesson. Sorry that it took me a while to look at it, not only did we work on a home remodel during the holidays, but I also got sick. Let me know what you think of my one suggestion, as it was not part of the original issue. Just something that stuck out to me as I was re-reading the entire lesson.

No problem. Hope you are feeling well now.

File: ruby_programming/basic_ruby/debugging.md (hunk 1 above)
Comment (Maheshkumar-novice) [reply ruby_programming/basic_ruby/debugging.md:132]: Yeah. It'll be useful. I'll add it.

Comment (Maheshkumar-novice) [reply ruby_programming/basic_ruby/debugging.md:132]: When your code executes and gets to `binding.pry`, it will open an IRB-like session in your terminal. You can then use that session to check the values of anything within the scope of where you included `binding.pry`. However, keep in mind that any code written *after* the `binding.pry` statement will not have been evaluated during the Pry session. 

For example, here `original_length` and `string_array` are in scope but not `unique_length` that is written after `binding.pry` and it's not evaluated yet. 

Thus, adding a `binding.pry` line in our code is similar to creating a breakpoint in JavaScript.


**Is this satisfies our need?**

Comment (rlmoser99) [reply ruby_programming/basic_ruby/debugging.md:132]: You are quick! I'd like to break up that long sentence into two, because it is slightly hard to follow. Maybe something like:

...are in scope. However, `unique_length` is not in scope, because it is written after `binding.pry` and has not been evaluated yet.

Comment (Maheshkumar-novice) [reply ruby_programming/basic_ruby/debugging.md]: Done. I've updated it.

**Review by rlmoser99 — APPROVED** [review]
This looks great! Thank you so much for taking the time to refactor this lesson!

**Comment by Maheshkumar-novice** [comment]
> This looks great! Thank you so much for taking the time to refactor this lesson!

Glad I helped. Thank you for guiding me and taking to time to do this.
